subprojects/gradle/deeplink-generator/src/main/kotlin/com/avito/deeplink_generator/ValidatePublicDeeplinksTask.kt

```diff
@@ -0,0 +1,79 @@
+package com.avito.deeplink_generator
+
+import com.avito.capitalize
+import com.avito.deeplink_generator.internal.parser.DeeplinkFileParser
+import com.avito.deeplink_generator.internal.validator.PublicDeeplinksValidator
+import com.avito.deeplink_generator.model.Deeplink
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Validates that deeplinks which are marked as public in code, also presented in build script, and vice versa.
+ *
+ * Deeplinks from code can be supplied directly with [publicDeeplinksFromCode] variable,
+ * or can be stored in file and supplied with [publicDeeplinksFromCodeFile].
+ */
+public abstract class ValidatePublicDeeplinksTask : DefaultTask() {
+
+    @get:Input
+    @get:Optional
+    public abstract val publicDeeplinksFromCode: SetProperty<Deeplink>
+
+    /**
+     * Input file with public deeplinks extracted from code.
+     * Supported format is specified in [DeeplinkFileParser].
+     */
+    @get:InputFile
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:Optional
+    public abstract val publicDeeplinksFromCodeFile: RegularFileProperty
+
+    /**
+     * Hint for user to fix their code if it does not contain deeplinks from build script.
+     *
+     * Example: `Please, add isPublic = true to @DeeplinkMeta annotation for all deeplinks listed above.`
+     */
+    @get:Input
+    @get:Optional
+    public abstract val codeFixHint: Property<String>
+
+    @get:Input
+    public abstract val defaultScheme: Property<String>
+
+    @get:Input
+    public abstract val publicDeeplinksFromBuildScript: SetProperty<Deeplink>
+
+    @TaskAction
+    public fun validate() {
+        val generatedDeeplinks = extractPublicDeeplinksFromCode()
+        val deeplinksFromBuildScript = publicDeeplinksFromBuildScript.get()
+        val codeFixHint = codeFixHint.orNull ?: ""
+        PublicDeeplinksValidator.validate(generatedDeeplinks, deeplinksFromBuildScript, codeFixHint)
+    }
+
+    private fun extractPublicDeeplinksFromCode(): Set<Deeplink> {
+        val publicDeeplinksFromCode = publicDeeplinksFromCode.orNull
+        val publicDeeplinksFromCodeFile = publicDeeplinksFromCodeFile.orNull?.asFile
+
+        return when {
+            publicDeeplinksFromCodeFile != null ->
+                DeeplinkFileParser.parse(publicDeeplinksFromCodeFile, defaultScheme.get())
+            publicDeeplinksFromCode != null -> publicDeeplinksFromCode
+            else -> error(
+                "You must supply either `publicDeeplinksFromCode` or `publicDeeplinksFromCodeFile` to run this task."
```

Comment:
> You must supply ... to run this task

If I see this message, I won't understand how to do it.
Let's mention here the extension.

Reply:
We're configuring inputs for this task not in extension, but in the task itself. 

publicDeeplinksFromCode / publicDeeplinksFromCodeFile may not exist when the extension is created. I'll try to rethink the way the links are supplied. Thanks for feedback.